### PR TITLE
bunch of fixes

### LINKED ISSUE #58: Error handling
Hi
why has a failed dispatch action the field `err` in the result payload?

```
function(err, result) {

}
```
`err = <error>`
`result = { err: <error> }`

if you pass an error and a result the payload looks like 

```
result: { err: <error>, <additional-result-> }
```

### PR DESCRIPTION
**Pull request by davidmarkclements**
closes #58

### CODE REVIEW

**Comment by davidmarkclements** [comment]
closes #57

**Comment by davidmarkclements** [comment]
close #35, closes #18

**Comment by davidmarkclements** [comment]
closes #36

**Comment by davidmarkclements** [comment]
closes #42 - 100% coverage, plus some important improvements to how tcp transport handles the econnrefused scenario

**Comment by davidmarkclements** [comment]
@mcollina pls review :bowtie:

**Comment by davidmarkclements** [comment]
cc @mcdonnelldean

File: packages/mu-router/index.js [hunk 1]
```diff
@@ -159,4 +113,89 @@ module.exports = function (opts) {
     })
     return result
   }
+
+  function pattern (message, cb) {
+    // we are routing an outbound message as there is a pattern attached to the message
+    var match = run.lookup(message.pattern)
```
Comment (mcdonnelldean) [inline packages/mu-router/index.js]: combine with line 148

**Comment by mcdonnelldean** [comment]
LGTM

File: packages/mu-router/index.js (hunk 1 above)
Comment (davidmarkclements) [inline packages/mu-router/index.js]: done

**Pull request merged**